roles/matrix-dimension/templates/config.yaml.j2

```diff
@@ -17,7 +17,7 @@ homeserver:
   # The URL that Dimension should use when trying to communicate with federated APIs on
   # the homeserver. If not supplied or left empty Dimension will try to resolve the address
   # through the normal federation process.
-  #federationUrl: "https://t2bot.io:8448"
+  federationUrl: "https://matrix-synapse:8448"
```

Comment:
My setup does not expose matrix to other servers, so I can't use https://{{ matrix_server_fqn_matrix }}:8448 here. And I think it would me more universal solution to use Docker's network, not external one.
But this configuration still gives me error:
```
Mar 08 21:49:11 srv-matrix-riot docker[31882]: Mar-8-2019 19:49:11.005 +00:00 info [Webserver] Incoming request: POST /api/v1/scalar/register
Mar 08 21:49:11 srv-matrix-riot docker[31882]: Mar-8-2019 19:49:11.006 +00:00 verbose [matrix] Cached federation URL for EXAMPLE.COM is https://matrix-synapse:8448
Mar 08 21:49:11 srv-matrix-riot docker[31882]: Mar-8-2019 19:49:11.007 +00:00 info [matrix] Doing federated API call: https://matrix-synapse:8448/_matrix/federation/v1/openid/userinfo
Mar 08 21:49:11 srv-matrix-riot docker[31882]: Mar-8-2019 19:49:11.011 +00:00 error [matrix] Error calling /_matrix/federation/v1/openid/userinfo
Mar 08 21:49:11 srv-matrix-riot docker[31882]: Mar-8-2019 19:49:11.012 +00:00 error [matrix] connect ECONNREFUSED 172.18.0.4:8448
```

Reply:
I agree, using the local URL is preferable (unless it causes some other trouble). Hopefully, as the comment says, `federationUrl` is only used by Dimension internally.

------------

Unless you have `matrix_synapse_federation_enabled: false` to completely disable federation, the `federation` listener is available on `http://matrix-synapse:8048` (note that this is `http`).

`https://matrix-synapse:8448` usually doesn't work, because Synapse usually does not do TLS termination by itself anymore (it's matrix-nginx-proxy that handles `matrix.DOMAIN:8448` and forwards it to `http://matrix-synapse:8048`). It's only if `matrix_synapse_tls_federation_listener_enabled: true` that `matrix-synapse:8448` is available.

In any case, if a server is federating, `http://matrix-synapse:8048` is always exposed, so that's the one we can bet on.

Reply:
Actually I had `matrix_synapse_federation_enabled: false`. Added validation rule for this.
`http://matrix-synapse:8048` works.

Reply:
That's a good idea!